### PR TITLE
fix regex in testURL

### LINKED ISSUE #171: Fix code scanning alert - Inefficient regular expression
Tracking issue for:
- [ ] https://github.com/Plant-for-the-Planet-org/planet-donations/security/code-scanning/2

### PR DESCRIPTION
**Pull request by sarveshpro**
Fixes #171

### CODE REVIEW

File: src/Layout/QueryParamContext.tsx [hunk 1]
```diff
@@ -153,15 +153,8 @@ export default function QueryParamProvider({ children }: any) {
...
-    const pattern = new RegExp(
-      "^(https?:\\/\\/)?" + // protocol
-        "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // domain name
-        "((\\d{1,3}\\.){3}\\d{1,3}))" + // OR ip (v4) address
-        "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // port and path
-        "(\\?[;&a-z\\d%_.~+=-]*)?" + // query string
-        "(\\#[-a-z\\d_]*)?$",
-      "i"
-    ); // fragment locator
+    const pattern = new RegExp(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g);
```
Comment (norbertschuler) [inline src/Layout/QueryParamContext.tsx:156]: Maybe you should escape `//` as `\/\/` at the end of the regex?

**Review by norbertschuler — APPROVED** [review]
Looks ok for me. Just one part of the regex might get some escaping?

File: src/Layout/QueryParamContext.tsx (hunk 1 above)
Comment (sarveshpro) [reply src/Layout/QueryParamContext.tsx]: Gives ES Lint error now
image

**Comment by norbertschuler** [comment]
This is on me, sorry, please revert your change:
```
/home/runner/work/planet-donations/planet-donations/src/Layout/QueryParamContext.tsx
  156:122  error  Unnecessary escape character: \/  no-useless-escape
  156:124  error  Unnecessary escape character: \/  no-useless-escape
```

File: src/Layout/QueryParamContext.tsx (hunk 1 above)
Comment (norbertschuler) [reply src/Layout/QueryParamContext.tsx]: This is on me, sorry, please revert your change.

**Comment by sarveshpro** [comment]
Let's merge this after Funding donations release is out, as this may require some additional testing

**Pull request merged**